### PR TITLE
Dry run concordium transactions first

### PR DESCRIPTION
**Pull request by abizjak**
## Purpose

Dry run Concordium state updates to check whether the operation has already completed, and to estimate energy.

## Checklist

- [ ] My code follows the style of this project.
- [ ] The code compiles without warnings.
- [ ] I have performed a self-review of the changes.
- [ ] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [ ] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: relayer/src/db.rs [hunk 1]
```diff
@@ -1132,9 +1132,9 @@ async fn insert_into_db(
                         };
                         let update = concordium_contracts::StateUpdate::Deposit(deposit);
                         // TODO estimate execution energy.
```
Comment (limemloh) [inline relayer/src/db.rs:1134]: I think you can remove the TODO now

File: relayer/src/concordium_contracts.rs [hunk 2]
```diff
@@ -207,12 +207,50 @@ impl BridgeManager {
     /// Construct a update transaction that can be sent to execute the provided
     /// [`StateUpdate`]. The expiry time of the transaction is set for 1d.
     ///
-    /// This **does not** send the transaction.
-    pub fn make_state_update_tx(
+    /// This **does not** send the transaction, but does dry run the update.
```
Comment (limemloh) [inline relayer/src/concordium_contracts.rs:210]: Consider explaining the Option in the return type

**Review by limemloh — APPROVED** [review]

File: relayer/src/concordium_contracts.rs (hunk 2 above)
Comment (abizjak) [reply relayer/src/concordium_contracts.rs:218]: Added.

File: relayer/src/db.rs (hunk 1 above)
Comment (abizjak) [reply relayer/src/db.rs]: Removed.

**Pull request merged**